Support for nested tokenPropertyName (modified)

eluciano11 opened the pull request:
This pull request contains the commits made by @thhermansen on #124 . It does not include the upgrade to ember and to ember-cli. Upgrade to ember and ember-cli 1.13.15 were made on #136.

thhermansen committed:
Decode payload with non ASCII chars correctly.

thhermansen committed:
Support nested token prop name in restore().

thhermansen committed:
refreshAccessToken supports nested token property name.

thhermansen committed:
Token authorizer supports nested token property name.

thhermansen committed:
Removed duplication: authenticate and refreshAccessToken.

thhermansen committed:
Double quotes to single quotes.

digia committed:
Add bug fix for JWT restore

JWT restore's refreshAccessToken should `resolve` with the data it's
provided, not with the original data. Restoring with the original data doesn't
update the token in the store.

eluciano11 committed:
Merge branch 'master' of https://github.com/eluciano11/ember-simple-auth-token

jpadilla commented on addon/authenticators/jwt.js:
```diff
@@ -158,24 +158,13 @@ export default TokenAuthenticator.extend({
-
     return new Ember.RSVP.Promise((resolve, reject) => {
-      const data = this.getAuthenticateData(credentials);
+      let data = this.getAuthenticateData(credentials);
```
Any reason why this can't be `const`?

jpadilla commented on addon/authenticators/jwt.js:
```diff
@@ -226,26 +215,14 @@ export default TokenAuthenticator.extend({
-    let data = {
-      [this.tokenPropertyName]: token
-    };
+    let data = this.makeRefreshData(token);
```
Any reason why this can't be `const`?

thhermansen committed:
Use const, instead of let, as we don't require any re-assigning.

thhermansen committed:
Set explicit number of assertions expected to be made.

thhermansen committed:
Fixed failing jwt authenticator test by ensure async test.

Seems like qunit did move on before the test was
completed, thus I did what
https://github.com/rwjblue/ember-qunit#async-example
documents on how to run async tests.

thhermansen committed:
Use Qunit's async() to wait for test to complete.

I did like that better than to use promise, as we
didn't have a function returning a promise in the
test this is simpler solution.

eluciano11 committed:
Merge pull request #2 from thhermansen/eluciano11-master

Fix to failing tests

eluciano11 committed:
Adds block comment to makeRefreshData with mod

Adds a block comment that explains what
makeRefreshData does and changes some variables
from let to const.

homu commented:
:pushpin: Commit 021c7d3 has been approved by `eluciano11`

<!-- @homu r=eluciano11 021c7d376167f5e5a4a261f1b5eb21456b52851a -->

homu commented:
:sunny: Test successful - [status](https://travis-ci.org/jpadilla/ember-simple-auth-token/builds/119010566)
